decidim-core/app/models/decidim/scope.rb

```diff
@@ -60,19 +60,19 @@ def forbid_cycles
     end
 
     def create_part_of
-      self[:part_of] = calculate_part_of
+      recalculate_part_of
       save if changed?
     end
 
     def update_part_of
-      self[:part_of] = calculate_part_of
+      recalculate_part_of
     end
 
-    def calculate_part_of
+    def recalculate_part_of
```

Comment:
I also find this name a bit misleading. I understand the change since before we were only calculating and not assigning, and now we're doing both. But "recalculate" seems to imply that it's been already calculated, but that's not the case for the "create" action.

I would remove the method and merge it into `update_part_of`, like this:

```ruby
def create_part_of
  update_part_of
  save if changed?
end

def update_part_of
  if parent
    part_of.clear.append(id).concat(parent.reload.part_of)
  else
    part_of.clear.append(id)
  end
end
```

What do you think?